src/Ecommerce.js

```diff
@@ -0,0 +1,161 @@
+        let newBrands;
+        if(index === -1){
+            newBrands = [...brandNames];
+            newBrands.push(name);
+            setBrandNames(newBrands);
+        }
+        else {
+            newBrands = [...brandNames]; 
+            newBrands = newBrands.filter(brand => {
+                return brand !== name;
+            });
+            setBrandNames(newBrands);
+        };
+
+        setCurrPage(1);
+    }
+
+    const sortByPrice = (option) => {
+        setSort(option);
+        setCurrPage(1);
+    }
+
+    const changeLayout = (val) => {
+        setProductsPerPage(3*val);
+        setCurrPage(1);
+    }
+
+    return (
+        <div>
+            <main className="wrapper">
+                <div>
+                <div className="card" style={{ width: "18rem" }}>
+                    <div className="card-header">Brands</div>
+                    <ul className="list-group list-group-flush">
+                    
+                        {brands ? brands.map(brand => {
+                            return <li key={brand.id} className="list-group-item">
+                                <input type="checkbox" name="brand" onChange={() => {filterByBrand(brand.name)}}/>{brand.name}
+                            </li>;
+                        }) : null}
+
+                    </ul>
+                </div>
+                <div className="card" style={{ width: "18rem" }}>
+                    <div className="card-header">Price</div>
+                    <ul className="list-group list-group-flush">
+                        <li className="list-group-item">
+                            <input type="radio" name="price" onChange={() => sortByPrice("Low to high")}/>Low to high
+                        </li>
+                        <li className="list-group-item">
+                            <input type="radio" name="price" onChange={() => sortByPrice("High to low")}/>High to low
+                        </li>
+                    </ul>
+                </div>
+                </div>
+                {filteredProducts ?
+                <div className="container">
+                    <div className="row">
+                        <div className="col-sm-5"></div>
+                        <div className="col-sm-3">Change Layout</div>
+                        <div className="col-sm-2"><button onClick={() => changeLayout(3)}>3 in row</button></div>
+                        <div className="col-sm-2"><button onClick={() => changeLayout(4)}>4 in row</button></div>
+                    </div>
+                    <div className="row">
+                        {filteredProducts.map(product => {
+                            return (
+                                <div key={product.id} className={productsPerPage === 9 ? "card col-4" : "card col-3"} onClick={() => { goToProduct(product.id) }}>
+                                    <img className="card-img-top" src={product.imageUrl} alt="Card image" />
```

Comment:
I think you can split this to component, also in all of the other places you did render items, because now this component is too big.